jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -356,13 +356,9 @@ ecma_builtin_number_prototype_object_value_of (ecma_value_t this_arg) /**< this
     if (ecma_object_get_class_name (obj_p) == LIT_MAGIC_STRING_NUMBER_UL)
     {
       ecma_property_t *prim_value_prop_p = ecma_get_internal_property (obj_p,
-                                                                       ECMA_INTERNAL_PROPERTY_PRIMITIVE_NUMBER_VALUE);
+                                                                       ECMA_INTERNAL_PROPERTY_ECMA_VALUE);
 
-      ecma_number_t *prim_value_num_p;
-      prim_value_num_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_number_t,
-                                                          ecma_get_internal_property_value (prim_value_prop_p));
-
-      return ecma_make_number_value (*prim_value_num_p);
+      return ecma_copy_value (ecma_get_internal_property_value (prim_value_prop_p));
```

Comment:
No assert here that `ecma_get_internal_property_value (prim_value_prop_p)` is indeed a number?